src/Cli/Command/AuthorizeCommand.php

```diff
@@ -53,38 +57,44 @@ protected function execute(InputInterface $input, OutputInterface $output)
         $client = $this->getClient();
         $domain = $input->getArgument('domain');
 
+        $solverName = strtolower($input->getOption('solver'));
+        if (!$this->getContainer()->has('challenge_solver.'.$solverName)) {
+            throw new \UnexpectedValueException(sprintf('The solver "%s" does not exists', $solverName));
+        }
+        /** @var SolverInterface $solver */
+        $solver = $this->getContainer()->get('challenge_solver.'.$solverName);
+
         $output->writeln(sprintf('<info>Requesting an authorization token for domain %s ...</info>', $domain));
-        $authorization = $client->requestAuthorization($domain);
 
-        $this->getRepository()->storeDomainAuthorizationChallenge($domain, $authorization);
+        $authorizationChallenges = $client->requestAuthorization($domain);
+        $authorizationChallenge = null;
+        foreach ($authorizationChallenges as $candidate) {
+            if ($solver->supports($candidate)) {
+                $authorizationChallenge = $candidate;
+                break;
+            }
+        }
+
+        if (null === $authorizationChallenge) {
+            throw new ChallengeNotSupportedException();
+        }
 
-        $this->output->writeln(sprintf(<<<'EOF'
+        $this->getRepository()->storeDomainAuthorizationChallenge($domain, $authorizationChallenge);
 
-<info>The authorization token was successfully fetched!</info>
+        $this->output->writeln('<info>The authorization token was successfully fetched!</info>');
+        $solver->solve($authorizationChallenge);
 
-Now, to prove you own the domain %s and request certificates for this domain, follow these steps:
+        $this->output->writeln(sprintf(
+<<<'EOF'
+<info>Then, you can ask to the CA to check the challenge!</info>
+    Call the <info>check</info> command to ask the server to check your URL:
 
-    1. Create a text file accessible on URL http://%s/.well-known/acme-challenge/%s
-       containing the following content:
-       
-       %s
-       
-    2. Check in your browser that the URL http://%s/.well-known/acme-challenge/%s returns
```

Comment:
I think you should keep this part in the SimpleHttpSolver, that's important to avoid issues. If you could also keep a numeroted list, that would be great too for a better display :) .